### ISSUE TITLE
Create Invoice Test Fail with Certain Timezones

### ISSUE THREAD

**Issue by enochtsang** [HospitalRun/hospitalrun-frontend#270]
I am using:
 - Linux Mint 17.1 Rebecca (64bit) (Should be similar enough to Ubuntu 14.04)
 - Google Chrome 47.0.2526.106 (Official Build) (64-bit)
 - ember 1.13.14
 - CouchDB 1.6.1 Vendor

Using a fresh install, the `create-invoice` test will fail with any time zone between -0500 and -1100. Time zones between -0600 and +1200 will pass. This is due to the test waiting for the visit date 11/1/2015 - 11/12/2015, but the visit date that appears is 10/31/2015 - 11/11/2015.

This will also cause 33 other tests to fail.

**Comment by enochtsang** [comment]
I tried changing the code to 
```javascript
    triggerEvent('.invoice-patient .tt-input', 'blur');

    var utcStartTime = moment.utc('2015-11-1 05:00:00').format('YYYY-MM-DD HH:mm:ss');
    var localStartTime  = moment.utc(utcStartTime).toDate();
    localStartTime = moment(localStartTime).format('M/D/YYYY');

    var utcEndTime = moment.utc('2015-11-12 05:00:00').format('YYYY-MM-DD HH:mm:ss');
    var localEndTime  = moment.utc(utcEndTime).toDate();
    localEndTime = moment(localEndTime).format('M/D/YYYY');

    var dateRange = localStartTime + ' - ' + localEndTime;

    waitToAppear('.invoice-visit option:contains(' + dateRange + ' (Admission)');
    andThen(function() {
      select('.invoice-visit', dateRange + ' (Admission)');
      fillIn('.external-invoice-no input', 'inv000002');
    });
```
In `invoices-test.js`, which got all tests to pass in all timezones in the browser, but the tests will still fail the same in the browser.

**Comment by jkleinsc** [comment]
@enochtsang I'm not sure I understand what you mean when you say: " all tests to pass in all timezones in the browser, but the tests will still fail the same in the browser."  Do the tests pass or fail?

As far as the UTC time, here is what is going on:  when a date that doesn't require a time is entered into the system, it gets entered as midnight of that day because javascript dates always contain a time.  Date/times are recorded in the system in the user's timezone/locale.  When the test data was created it was created in Eastern Standard Time which is -5 UTC, thus the utc time of 05:00:00.  Does that make sense?  Also, just to restate, times in the system are stored as UTC/javascript dates, but what that UTC value is will be relative to the user's timezone.

**Comment by tangollama** [comment]
Adding Help Wanted for assistance in duplicating the issue for more information.

**Comment by jkleinsc** [comment]
This was fixed via pr #340
